### PR TITLE
Support rich assertions.

### PR DESCRIPTION
**Pull request by wsx-antithesis**

### CODE REVIEW

File: sdk/src/main/java/com/antithesis/sdk/internal/Guidance.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                this.emit();
+            }
+        } else {
+            this.emit();
+        }
+    }
+
+    private void emit() {
+        ObjectNode guidacnNode = MAPPER.createObjectNode();
+        guidacnNode.put("antithesis_guidance", MAPPER.valueToTree(this));
```
Comment (herzogp) [inline sdk/src/main/java/com/antithesis/sdk/internal/Guidance.java]: please repair `guidanceNode` spelling

**Review by herzogp — CHANGES_REQUESTED** [review]
Otherwise, looks good.  I don't know about the precision on all possible integral values that can be represented in a double - might result in some inaccuracies.  Nice job handling +/-Infinity.  Is NaN also handled somehow?

**Comment by wsx-antithesis** [comment]
> I don't know about the precision on all possible integral values that can be represented in a double - might result in some inaccuracies.

I followed the Rust SDK behavior (which admittedly can be improved) here. The rational of converting all numeric values to double is that eventually they are serialized & deserialize through JSON, which becomes double in the fuzzer/notebook. The precision issue may result in a few more guidance being reported (when they are rounded to the watermark), but we'll never incorrectly omit one that should have been reported.

> Is NaN also handled somehow?

Good catch! I updated the NaN handling logic so that NaN values are always reported, but they never affect the water mark, so future normal values can still go through. This is the same behavior as the Rust SDK, and I find it sensible.

File: sdk/src/main/java/com/antithesis/sdk/internal/Guidance.java (hunk 1 above)
Comment (wsx-antithesis) [reply sdk/src/main/java/com/antithesis/sdk/internal/Guidance.java]: Thanks, now fixed!

**Pull request merged**